### ISSUE TITLE
Promotion Success / Process Failed

### ISSUE THREAD

**Issue by darkfrog26** [xerial/sbt-sonatype#276]
**Describe the bug**
After I get the message "The promote stage is in progress", it starts showing:
`Execution failed: [404: Not Found] Request failed: {"errors":[{"id":"*","msg":"No activity for repository: comoutr-1003"}]}. Retrying in 15.00 sec.  - (SonatypeClient.scala:172)`

Perpetually. It would appears as though the promotion was successful, but the plugin never finishes because it just keeps getting the "No activity" response.

**How to reproduce the issue**
It doesn't always happen, but it is frequently happening ever since I migrated to "https://s01.oss.sonatype.org/service/local"

- sbt version: 1.6.1
- sbt-sonatype version: 3.9.10
- Show your build settings:
```
$ sbt
> sonatypeProfileName
com.outr
> sonatypeRepository
https://s01.oss.sonatype.org/service/local
```
- GitHub repository: [(Having a GitHub repository for reproducing the issue would be helpful)](https://github.com/outr/scribe)

**Additional context**
Just released 3.6.6 and it happened with it, but I believe it was successfully released.

**Comment by mkurz** [comment]
More people experienced this: https://github.com/sbt/sbt-ci-release/issues/223

**Comment by xerial** [comment]
I think we need to confirm two things:
- A: Is sbt-sonatype is accessing the endpoints in https://s01.oss.sonatype.org This configuration is extracted here https://github.com/xerial/sbt-sonatype/blob/245cf62923595ae9605d80bc3c5bf8942538799c/src/main/scala/xerial/sbt/Sonatype.scala#L416
- B: Is there any change in the behavior between s01.oss.sonatype.org and oss.sonatype.org about the closing endpoint.

Unfortunately, I don't have any s01 sonatype account, so help for confirming these two points is appreciated.

**Comment by pomadchin** [comment]
Got the same issue after upgrading `1.5.7` => `1.5.10`, https://github.com/CartoDB/analytics-toolbox-databricks/runs/5159176823?check_suite_focus=true

**Comment by pomadchin** [comment]
I had a chance to look into it a bit: https://github.com/xerial/sbt-sonatype/compare/3.9.7...3.9.11
The only suspisious PR in this list is https://github.com/xerial/sbt-sonatype/pull/242

There is indeed a regression, which is solved now by having a `ThisBuild / sonatypeCredentialHost := "s01.oss.sonatype.org"` defined in the build.sbt file. 

TlDR; perproject `sonatypeCredentialHost` gots overrided and is not enough to have, that should be set globally only.

A [tiny proof](https://github.com/CartoDB/analytics-toolbox-databricks/commit/cd9089d9848a5c718d8257275a8b3cbda2ba646c) with [the green release](https://github.com/CartoDB/analytics-toolbox-databricks/runs/5161385195?check_suite_focus=true#step:5:113).

I am not sure if that is a desired behavior.

**Comment by xerial** [comment]
@pomadchin Thanks.

It seems @darkfrog26 also has this setting, but experiencing the same error occasionally:
```
ThisBuild / sonatypeCredentialHost := "s01.oss.sonatype.org"
```

To identify the cause, adding some info log to see which repository URLs are actually used around here would be the next step: https://github.com/xerial/sbt-sonatype/blob/245cf62923595ae9605d80bc3c5bf8942538799c/src/main/scala/xerial/sbt/Sonatype.scala#L416-L418

**Comment by pomadchin** [comment]
@xerial hmmm the only difference with my build is that I have it set both per project and globally ._.' what a tricky behavior to debug.

**Comment by xerial** [comment]
It looks like the issue of configuration scopes. I guess depending on the context within sbt, extracting project settings via Project.extract(state) references unexpected projects. 
https://github.com/xerial/sbt-sonatype/blob/245cf62923595ae9605d80bc3c5bf8942538799c/src/main/scala/xerial/sbt/Sonatype.scala#L412

An approach would be changing sonatypeXXX commands into an sbt task definition or finding a way to get a consistent reference to the root project when calling `Project.extract(state)`

**Comment by xerial** [comment]
Probably I've found the cause:
https://github.com/xerial/sbt-sonatype/blob/245cf62923595ae9605d80bc3c5bf8942538799c/src/main/scala/xerial/sbt/Sonatype.scala#L68

This setting is not referencing `ThisBuild / sonatypeCredentialHost`, so the setting might vary depending on the context project. If that's the case, your log might show in the failed deployment:
```
 info [SonatypeService] sonatypeRepository  : https://oss.sonatype.org/service/local
```

If s01.oss.sonatype.org is set in sonatypeCredentialHost, the url should be https://s01.oss.sonatype.org/service/local

**Comment by darkfrog26** [comment]
@xerial, awesome! When will this be released?

**Comment by darkfrog26** [comment]
All of my open-source projects are now on s01, so I'll drop a message if I see this again. Thanks, @xerial!

**Comment by simao** [comment]
I am getting the same problem since I upgraded to 3.9.12:

```
[MISSING_CREDENTIAL] No credential is found for oss.sonatype.org. Prepare ~/.sbt/(sbt_version)/sonatype.sbt file.

```

Thanks

**Comment by darkfrog26** [comment]
@simao you need to update your `sonatype.sbt` file to reference `s01.oss.sonatype.org` instead of `oss.sonatype.org`.

**Comment by xerial** [comment]
We need to release a new version, which fix this regression

**Comment by albuch** [comment]
I still get the original error of this issue (error after promotion) with `v3.9.12`.
My build configuration: https://github.com/albuch/sbt-dependency-check/blob/main/build.sbt

**Comment by xerial** [comment]
Released sbt-sonatype 3.9.13, which reverted #276.

**Comment by xerial** [comment]
This issue is related to https://github.com/xerial/sbt-sonatype/issues/309
